src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ReactNode, RefObject, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { IconAlertCircle, IconCross } from 'hds-react';
+import { PrimaryButton, SecondaryButton } from '../button/Button';
+import './Modal.scss';
+
+const ModalBackDrop = (): JSX.Element => <div className="modal-backdrop" />;
```

Comment:
Im not sure if extracting this ModalBackDrop here is a good idea? It increases the jumps reader has to do while reading the code, yet it's so small in size that I feel we could maybe just inline it directly to the code?

Reply:
I will inline this.